### PR TITLE
feat: #5 Add README.md

### PR DESCRIPTION
**Pull request by downzed**

### CODE REVIEW

**Review by josephi — COMMENTED** [review]
Added comments

File: README.md [hunk 1]
```diff
@@ -1 +1,94 @@
...
+- **Customizable Prompts**: Choose from various prompts for code completion, refactoring, and documentation.
+- **Extensible**: Easily extend the functionality by adding new prompts or modifying existing ones.
+
+## Installation
+
+### Using Packer
+
+```lua
+use {
+  'downzed/die-schnecke.nvim',
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:23]: That's how plugins are installed in `lazy.nvim`.

File: README.md [hunk 2]
```diff
@@ -1 +1,94 @@
...
+Plug 'downzed/die-schnecke.nvim'
+lua require('die-schnecke').setup{}
+```
+
+## Usage
+
+### Commands
+
+- `:OpenDieSchnecke` - Open the Die Schnecke interface.
+- `:OpenDieSchneckeWithSelectedCode` - Open Die Schnecke with the currently selected code with a preview.
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:42]: The Snail.
For two reasons:
1. German word I learned 
2. My learning/developing process, specifically for this plugin. (Not rushing)

File: README.md [hunk 3]
```diff
@@ -1 +1,94 @@
+
+> Based on the provided code from this repo, and ChatGpt-4o (so you might find mistakes):
+
+---
+
 # die-shchnecke.nvim
+
+die-shchnecke.nvim: (personal, slow) Neovim plugin designed to enhance your coding experience by integrating with the LLaMA language model for code completion, refactoring, and documentation generation.
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:8]: You're right, should be more specific here

File: README.md [hunk 4]
```diff
@@ -1 +1,94 @@
...
+    stream = true,
+    port   = "11434"
+  },
+}
+```
+
+## API
+
+die-schnecke.nvim will expose several functions that you can use in your configuration or other plugins.
+sorry for the inconvenience.
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:72]: In the lua plugin for neovim, whenever I expose a certain api, you can use it in your own configuration. (e.g. `require"die-schnecke".exposed_api()`)

File: README.md [hunk 5]
```diff
@@ -1 +1,94 @@
...
+
+- Expose relevant API functions, and or classes
+- Code completion engine
+- Prompts:
+    - Specific model for specific "task"
+    - Function calling
+
+## Known Issues (sorry)
+- win+buf regenaration
+- running jobs in the background
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:85]: The plugin's job of the ollama server.

File: README.md [hunk 6]
```diff
@@ -1 +1,94 @@
...
+## TODO
+
+- Expose relevant API functions, and or classes
+- Code completion engine
+- Prompts:
+    - Specific model for specific "task"
+    - Function calling
+
+## Known Issues (sorry)
+- win+buf regenaration
```
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:84]: I think so.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (josephi) [reply README.md:23]: so it pull the plugin directly from github? how it knows that it is stored in github? is there a neovim registry like npm?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (josephi) [reply README.md:42]: so explain in your redme what the name means.. and how it is spelled

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (josephi) [reply README.md:72]: but why you are sorry?

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (josephi) [reply README.md:85]: so explain it

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (downzed) [reply README.md:23]: Not sure about the registry.. every plugin manager works differently.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (josephi) [reply README.md:46]: good

**Pull request merged**